File: leetcode/Daily/2460.ApplyOperationsArray/solution.py
# -*- encoding: utf-8 -*-
"""
@Introduce  : 
@File       : solution.py
@Author     : ryrl
@Email      : ryrl970311@gmail.com
@Time       : 2025/03/01 16:34
@Description: 
"""

from typing import List

class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:

        n = len(nums)
        res = []

        for idx in range(n - 1):

            if nums[idx] == nums[idx + 1] and nums[idx] != 0:
                nums[idx] *= 2
                nums[idx + 1] = 0
        
        for idx in range(n):
            if nums[idx] != 0:
                res.append(nums[idx])
        
        for _ in range(n - len(res)):

            res.append(0)
        return res

File: leetcode/Daily/2460.ApplyOperationsArray/solution.rs
// @Introduce  :
// @File       : solution.rs
// @Author     : ryrl
// @Email      : ryrl970311@gmail.com
// @Time       : 2025/03/01 19:11
// @Description:

impl Solution {
    fn apply_operations(nums: Vec<i32>) -> Vec<i32> {
        let mut nums = nums;
        let n = nums.len();

        for idx in 0..n - 1 {
            if nums[idx] == nums[idx + 1] && nums[idx] != 0 {
                nums[idx] *= 2;
                nums[idx + 1] = 0;
            }
        }

        let mut res: Vec<i32> = nums.into_iter().filter(|&x| x != 0).collect();
        res.resize(n, 0);
        res
    }
}
